crates/hir-ty/src/tests/method_resolution.rs

```diff
@@ -8,6 +8,7 @@ use super::{check_infer, check_no_mismatches, check_types};
 fn infer_slice_method() {
     check_types(
         r#"
+//- /core.rs crate:core
```

Comment:
Another change is that we assume crates other than the sysroot cannot define incoherent impls (even rustc crates do not use this possibility, and it's an internal feature). This helps narrowing the search space, but it also means tests defining incoherent impls need to pretend to be core or std.